pkg/collector/collector.go

```diff
@@ -30,16 +29,39 @@ const (
 
 const cancelCheckTimeout time.Duration = 500 * time.Millisecond
 
+// EventType represents the type of events emitted by the collector
+type EventType uint32
+
+const (
+	// CheckRun is emitted when a check is added to the collector
+	CheckRun EventType = iota
+	// CheckStop is emitted when a check is stopped and removed from the collector
+	CheckStop
+)
+
+// EventReceiver represents a function to receive notification from the collector when running or stopping checks.
+type EventReceiver func(checkid.ID, EventType)
```

Comment:
The issue is that the collector requires inventory to trigger refresh and inventory requires the collector to pull data about each check. We used to have a small hack where inventory would redefine an interface for the collector to break the cycling dependencies but that no longer works with FX.

The real solution would be for the `inventory-checks` metadata provider to be part of the collector (like `agent-checks` provider). This will be done once the collector is a component.